src/libcore/ffi.rs

```diff
@@ -0,0 +1,143 @@
+// Copyright 2018 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+//! Implementation of a `va_list`.
+
+#[cfg(any(all(not(target_arch = "aarch64"), not(target_arch = "powerpc"),
+              not(target_arch = "x86_64")),
+          windows))]
+#[repr(transparent)]
+#[derive(Debug, Clone, Copy)]
+/// Basic implementation of a `va_list`.
+struct VaListImpl {
+    _dummy: u8
+}
```

Comment:
Does it work with `#[repr(C)]` instead of `#[repr(transparent)]`?
And if so, can you also try `extern { type VaListImpl; }`?
(that would be ideal because it can't be misused like `_dummy` can.

Reply:
Although then the uninitialized in `va_copy` wouldn't work as well - I'm assuming the uninitialized `VaListImpl` in this case is unused, and LLVM overwrites the pointer itself?

Reply:
Maybe you could have a method on `VaListImpl` that takes `&mut Self` (in the other cases) or `&()` (in this case), returning `&mut Self` in both cases, and `&mut uninitialized_variable` is passed to it from `copy`.

In this case, you'd return `&mut *(1 as *mut Self)` or something like that - a dummy pointer that could be overwritten later.

Reply:
> Does it work with #[repr(C)] instead of #[repr(transparent)]?
And if so, can you also try extern { type VaListImpl; }?
(that would be ideal because it can't be misused like _dummy can.

Trying this at the moment.

> Maybe you could have a method on VaListImpl that takes &mut Self (in the other cases) or &() (in this case), returning &mut Self in both cases, and &mut uninitialized_variable is passed to it from copy.

I think this should work. I don't think it should need to take any parameters though, since `self` should have no impact on the uninitialized output.

Reply:
Well, in all other cases you need to return a reference to an uninitialized memory location, *but the pointer* needs to be initialized. That's why the argument.